tools/cldr-code/src/test/java/org/unicode/cldr/unittest/TestTimeCycle.java

```diff
@@ -0,0 +1,184 @@
+            System.out.println(
+                    "testJjMapping "
+                            + localeID
+                            + "\t["
+                            + jPattern
+                            + "] ["
+                            + jPatSkeleton
+                            + "] ["
+                            + shortPattern
+                            + "] ["
+                            + shortPatSkeleton
+                            + "]");
+            for (char timeCycleChar : timeCycleChars) {
+                boolean has1 = jPatSkeleton.indexOf(timeCycleChar) >= 0;
+                boolean has2 = shortPatSkeleton.indexOf(timeCycleChar) >= 0;
+                if (has1 && !has2) {
+                    String calType = USE_ICU_DATA ? dfmt.getCalendar().getType() : calendarType;
+                    failOrLog(
+                            ++failureNumber
+                                    + " locale "
+                                    + localeID
+                                    + ", calendar "
+                                    + calType
+                                    + ", expected "
+                                    + timeCycleChar
+                                    + " to occur in both patterns "
+                                    + jPattern
+                                    + " and "
+                                    + shortPattern);
+                }
+            }
+        }
+    }
+
+    private String getCldrJPatternForLocale(
+            Map<String, PreferredAndAllowedHour> timeData,
+            String localeID,
+            Map<String, String> likelySubtags) {
+
+        PreferredAndAllowedHour prefAndAllowedHr = timeData.get(localeID);
+        if (prefAndAllowedHr == null) {
+            LocaleIDParser lp = new LocaleIDParser();
+            String region = lp.set(localeID).getRegion();
+            if (region == null || region.isEmpty()) {
+                String loc2 = likelySubtags.get(localeID);
+                if (loc2 != null && !loc2.isEmpty()) {
+                    region = lp.set(loc2).getRegion();
+                    System.out.println(
+                            "getCldrJPatternForLocale "
+                                    + localeID
+                                    + " loc2 = "
+                                    + loc2
+                                    + " region = "
+                                    + region);
+                }
+            }
+            prefAndAllowedHr = timeData.get(region);
+            if (prefAndAllowedHr == null) {
+                System.out.println(
+                        "getCldrJPatternForLocale "
+                                + localeID
+                                + " get (region "
+                                + region
+                                + ") null, falling back to 001");
+                prefAndAllowedHr = timeData.get("001" /* world */);
```

Comment:
fallback to 001 happens for az_Cyrl, az_Latn, bal_Arab, bal_Latn, be_TARASK, ..., altogether 53 locales. Strangely not the same as the 53 failing locales, though there is some overlap, such as for el_POLYTON